meteor/lib/lib.ts

```diff
@@ -374,7 +374,24 @@ export function cacheResult<T>(name: string, fcn: () => T, limitTime: number = 1
 	}
 	const cache = cacheResultCache[name]
 	if (!cache || cache.ttl < Date.now()) {
-		const value = fcn()
+		const value: T = fcn()
+		cacheResultCache[name] = {
+			ttl: Date.now() + limitTime,
+			value: value,
+		}
+		return value
+	} else {
+		return cache.value
+	}
+}
+/** Cache the result of function for a limited time */
+export async function cacheResultAsync<T>(name: string, fcn: () => Promise<T>, limitTime: number = 1000): Promise<T> {
+	if (Math.random() < 0.01) {
+		Meteor.setTimeout(cleanOldCacheResult, 10000)
```

Comment:
Just a side question: is this better/more performant than having a single interval just run the cache cleanup? Is the point here to have a bit of fuzzyness for the cleanup?

Reply:
Yeah, a setInterval would probably work just as fine. 
This might make `cleanOldCacheResult` to not run unnessesarily when idle, but really it doesn't matter, brcause when idle all of the cache is likely to have been emptied, so `cleanOldCacheResult` would take no time at all to run.